zilencer/lib/remote_counts.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+
+from django.db import connection
+from django.utils.timezone import now as timezone_now
+from psycopg2.sql import SQL, Literal
+
+from zilencer.models import RemoteInstallationCount, RemoteZulipServer
+
+
+class MissingDataError(Exception):
+    pass
+
+
+def compute_max_monthly_messages(remote_server: RemoteZulipServer) -> int:
+    # Calculate the maximum amount of messages that the server had within a month.
+    # out of the last 3 months.
+
+    # Filtering on property is a placeholder here; it
+    # currently doesn't change the result. Note that we can't
+    # filter on the actual property we care about
+    # ('messages_sent:message_type:day'), because zeros are not
+    # recorded.
+    # For non-legacy servers, we probably want to distinguish not uploading
+    # data vs. uploading minimal billing data (and thus not qualified for the
+    # Starter plan) vs. uploading all the data.
+    if not RemoteInstallationCount.objects.filter(
+        server=remote_server,
+        property="active_users_audit:is_bot:day",
```

Comment:
@alya Your query used `active_user:is_bot:day`, but `active_users_audit:is_bot:day` should be fine too? (it's easier to generate test data for this one I think - and it sounds like we consider this the primary user count stat - https://github.com/zulip/zulip/blob/6116679a9537bfc703fd9b50ea1057f9510a47a4/analytics/lib/counts.py#L754-L756)

Reply:
Either way, shouldn't really matter since it's just a placeholder for "the server has recently submitted their user/message/etc. analytics data", right?

Reply:
Yeah, just a placeholder here, so shouldn't make any difference.

Reply:
Yes, I've double-checked that is the value we show on graphs, so yeah this sounds correct.